### PR TITLE
palette

### PR DESCRIPTION
**Pull request by jiangzhengwenjz**

### CODE REVIEW

File: src/palette.c [hunk 1]
```diff
@@ -0,0 +1,987 @@
...
+
+// not used
+static u8 GetPaletteNumByUid(u16 uid)
+{
+    u8 i;
+
+    for (i = 0; i < 16; ++i)
+        if (sPaletteStructs[i].base->uid == uid)
+            return i;
+    return 16;
```
Comment (PikalaxALT) [inline src/palette.c]: Ascribe a define to this - either `NUM_PALETTE_STRUCTS` (and then use that in the declaration of `sPaletteStructs`) or `NELEMS(sPaletteStructs)`.

File: src/palette.c [hunk 2]
```diff
@@ -0,0 +1,987 @@
...
+    if (gPaletteFade.active)
+    {
+        return FALSE;
+    }
+    else
+    {
+        gPaletteFade.deltaY = 2;
+        if (delay < 0)
+        {
+            gPaletteFade.deltaY += (delay * -1);
```
Comment (PikalaxALT) [inline src/palette.c]: you really have to question what gf was smoking here (i expect there was a similar comment in ruby and emerald)

File: src/palette.c [hunk 3]
```diff
@@ -0,0 +1,987 @@
...
+        if (selectedPalettes & 1)
+        {
+            u8 i;
+
+            for (i = 0; i < 16; ++i)
+            {
+                struct PlttData *data = (struct PlttData *)&gPlttBufferFaded[paletteOffset + i];
+                data->r += r;
+                data->g += g;
+                data->b += b;
```
Comment (PikalaxALT) [inline src/palette.c]: Interesting that this isn't clamped at all.

File: src/palette.c (hunk 1 above)
Comment (jiangzhengwenjz) [reply src/palette.c]: Now  NUM_PALETTE_STRUCTS is there

**Pull request merged**